tests/main_test.bal

```diff
@@ -0,0 +1,283 @@
+    } else {
+        test:assertFail(msg = updateContactPhoto.message());
+    }
+}
+
+@test:Config {dependsOn: [testUpdateContactPhoto]}
+function testDeleteContactPhoto() {
+    log:print("Running Delete Contact Photo Test");
+    var deleteContactPhoto = googleContactClient->deleteContactPhoto(contactResourceName);
+    if (deleteContactPhoto is ()) {
+        test:assertTrue(true, msg = "Delete Contact Photo Failed");
+    } else {
+        test:assertFail(msg = deleteContactPhoto.message());
+    }
+}
+
+@test:Config {dependsOn: [testCreateContact, testGetContact, testDeleteContactPhoto]}
+function testDeleteContact() {
+    log:print("Running Delete Contact Test");
+    runtime:sleep(10);
+    var deleteContact = googleContactClient->deleteContact(contactResourceName);
+    if (deleteContact is ()) {
+        test:assertTrue(true, msg = "Delete Contact Failed");
+    } else {
+        test:assertFail(msg = deleteContact.message());
+    }
+}
+
+string contactGroupResourceName = "";
+
+@test:Config {}
+function testCreateContactGroup() {
+    log:print("Running Create Contact Group Test");
+    string[] readGroupFields = ["name", "clientData", "groupType", "metadata"];
+    var createContactGroup = googleContactClient->createContactGroup("TestContactGroup", readGroupFields);
+    if (createContactGroup is ContactGroup) {
+        log:print(createContactGroup.toString());
+        contactGroupResourceName = createContactGroup.resourceName;
+        test:assertTrue(true, msg = "Creating Contact Group Failed");
+    } else {
+        test:assertFail(msg = createContactGroup.message());
+    }
+}
+
+@test:Config {dependsOn: [testCreateContactGroup]}
+function testGetContactGroup() {
+    log:print("Running Get Contact Group Test");
+    runtime:sleep(10);
+    var getContactGroup = googleContactClient->getContactGroup(contactGroupResourceName);
+    if (getContactGroup is ContactGroup) {
+        log:print(getContactGroup.toString());
+        contactGroupResourceName = getContactGroup.resourceName;
+        test:assertTrue(true, msg = "Fetching Contact Group Failed");
+    } else {
+        test:assertFail(msg = getContactGroup.message());
+    }
+}
+
+@test:Config {dependsOn: [testCreateContactGroup]}
+function testbatchGetContactGroup() {
+    log:print("Running BatchGet Contact Group Test");
+    runtime:sleep(10);
+    string[] resourceNames = [contactGroupResourceName];
+    var batchGetContactGroup = googleContactClient->batchGetContactGroup(resourceNames);
+    if (batchGetContactGroup is ContactGroupBatch) {
+        log:print(batchGetContactGroup.toString());
+        test:assertTrue(true, msg = "Batch Get Contact Group Failed");
+    } else {
+        test:assertFail(msg = batchGetContactGroup.message());
+    }
+}
+
+@test:Config {dependsOn: [testCreateContactGroup]}
+function testListContactGroup() {
+    log:print("Running List Contact Group Test");
+    runtime:sleep(10);
```

Comment:
Why are we doing this? Any reason?

Reply:
I removed from this place. Resource creation take small time. that is why added in other places